tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2022.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+
+package e2e
+
+import (
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+// It's quite possible these integration tests fail due to maven
```

Comment:
You could expose a `Validate` function that wraps all this logic and just test that directly, substituting a fake http server. I know that's not exactly e2e, though.

Reply:
think I might leave it like this for now